### PR TITLE
CNV#27695: Update RN known issues and bug fixes for 4.15

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.15

Issue: [CNV-27695](https://issues.redhat.com//browse/CNV-27695)

Link to docs preview: https://71598--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/release_notes/virt-4-15-release-notes#virt-4-15-bug-fixes

QE review: N/A

Additional information:

### CODE REVIEW

**Review by phoracek — APPROVED** [review]
network LGTM

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 1]
```diff
@@ -182,6 +182,13 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
 
 [id="virt-4-15-bug-fixes"]
 == Bug fixes
+* Previously, EFI pipelines failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])
```
Comment (dankenigsberg) [inline virt/release_notes/virt-4-15-release-notes.adoc:185]: @sjhala-ccs @dominikholler this is old news, but I don't know what are `EFI pipelines`. Can a URL be added?

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 2]
```diff
@@ -182,6 +182,13 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
+
+* Previously, if you simultaneously cloned more than 1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```
Comment (dankenigsberg) [inline virt/release_notes/virt-4-15-release-notes.adoc:187]:
```suggestion
* Previously, if you simultaneously cloned more than ~1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```

@aglitke I think that it's not about 1000 precisely

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 3]
```diff
@@ -182,6 +182,13 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
+
+* Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```
Comment (dankenigsberg) [inline virt/release_notes/virt-4-15-release-notes.adoc:189]:
```suggestion
* Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```
@phoracek I think this should be revised to refer to GUI and ease of use.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 4]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
-//4.14 leave in per Miguel Duarte de Mora Barroso
-* If your {product-title} cluster uses OVN-Kubernetes as the default Container Network Interface (CNI) provider, you cannot attach a Linux bridge or bonding device to a host's default interface because of a change in the host network topology of OVN-Kubernetes. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1885605[*BZ#1885605*])
-** As a workaround, you can use a secondary network interface connected to your host, or switch to the OpenShift SDN default CNI provider.
 
-//CNV-33789: Cannot SSH into VM over NodePort and Console's FQDN when using OVNKubernetes networking
-* You cannot SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2165895[*BZ#2165895*])
+//[discrete]
+//[id="virt-4-15-ki-networking"]
+//==== Networking
+//CNV-10590 - This issue can be removed for 4.15 when CNV-13680 RN is added.
```
Comment (dankenigsberg) [inline virt/release_notes/virt-4-15-release-notes.adoc:208]: @sjhala-ccs why is this commented out and not removed? (really asking, no idea)

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 5]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
-//BZ2247593 from Ying
-* Live migration cannot be enabled for a virtual machine instance (VMI) after a hotplug volume has been added and removed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055595[*BZ#2247593*])
 
 [discrete]
 [id="virt-4-15-ki-virtualization"]
 ==== Virtualization
-//4.14 Leave in per Stu
-* {VirtProductName} links a service account token in use by a pod to that specific pod. {VirtProductName} implements a service account volume by creating a disk image that contains a token. If you migrate a VM, then the service account volume becomes invalid. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2037611[*BZ#2037611*])
+//New for 4.15
+* CPU hot unplug for Windows VMs is not supported and results in an error.
```
Comment (dankenigsberg) [inline virt/release_notes/virt-4-15-release-notes.adoc:242]: This may be misleading - we don't yet support any kind of cpu hotplug. @stu-gott

**Review by dankenigsberg — CHANGES_REQUESTED** [review]
partial review

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 4 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:208]: Was planning to remove it after I added in the release note for CNV-13680. Sorry about the confusion.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 6]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
+
+* When adding a virtual Trusted Platform Module (vTPM) device to a Windows VM, the BitLocker Drive Encryption system check passes even if the vTPM device is not persistent. This is because a vTPM device that is not persistent stores and recovers encryption keys using ephemeral storage for the lifetime of the `virt-launcher` pod. When the VM migrates or is shut down and restarts, the vTPM data is lost. (link:https://issues.redhat.com/browse/CNV-36448[*CNV-36448*])
+
+
+//Looks like this is still unresolved for 4.15
+* {VirtProductName} links a service account token in use by a pod to that specific pod. {VirtProductName} implements a service account volume by creating a disk image that contains a token. If you migrate a VM, then the service account volume becomes invalid. (link:https://issues.redhat.com/browse/CNV-33835[*CNV-33835*])
 ** As a workaround, use user accounts rather than service accounts because user account tokens are not bound to a specific pod.
 
-// Leave in for 4.14 per Stu; remove from 4.15 RNs and move content to CNV docs, per Avital
+//4.15 Check with Stu if this needs to be moved to CNV docs
```
Comment (sjhala-ccs) [inline virt/release_notes/virt-4-15-release-notes.adoc:252]: @stu-gott Do we still need to keep the below known issue in the release notes?

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:252]: The release note still applies. We are currently working toward virtiofs support in KubeVirt, which is the precursor to addressing this. Rough ETA for fixing this issue is forecast as 4.18+. If it's a problem to leave a RN in for that long we could consider moving this to somewhere in the documentation, but we do plan to fix it once we're able.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 7]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
 [discrete]
 [id="virt-4-15-ki-storage"]
 ==== Storage
-//4.14 Leave in per Adam
-* In some instances, multiple virtual machines can mount the same PVC in read-write mode, which might result in data corruption. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1992753[*BZ#1992753*])
+//4.15 Check with Alex Kalenyuk. Issue is closed as "Not a bug"
+* If you use Portworx as your storage solution on AWS and create a VM disk image, the created image might be smaller than expected due to the filesystem overhead being accounted for twice. (link:https://issues.redhat.com/browse/CNV-32695[*CNV-32695*])
+** As a workaround, you can manually expand the Persistent Volume Claim (PVC) to increase the available space after the initial provisioning process completes.
+
+//4.15 Check with Adam. Issue is closed as "Won't do" but is still a known issue
```
Comment (sjhala-ccs) [inline virt/release_notes/virt-4-15-release-notes.adoc:228]: @aglitke Should we still keep this issue for 4.15?

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 8]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
+* In some instances, multiple virtual machines can mount the same PVC in read-write mode, which might result in data corruption. (link:https://issues.redhat.com/browse/CNV-13500[*CNV-13500*])
 ** As a workaround, avoid using a single PVC in read-write mode with multiple VMs.
 
-//4.14 Leave in per Adam and Stu; will be doc'd in 4.15
-* If you clone more than 100 VMs using the `csi-clone` cloning strategy, then the Ceph CSI might not purge the clones. Manually deleting the clones might also fail. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2055595[*BZ#2055595*])
+//4.15 Have asked Dominik Holler if we still need to keep this
```
Comment (sjhala-ccs) [inline virt/release_notes/virt-4-15-release-notes.adoc:232]: @dominikholler Can you please confirm if we need to keep this issue for 4.15?

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 6 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:252]: Thanks for the update. We will leave the RN as is.

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 1 above)
Comment (dominikholler) [reply virt/release_notes/virt-4-15-release-notes.adoc:185]:
```suggestion
* Previously, the pipeline windows-efi-installer failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])
```

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 5 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:242]: @stu-gott Thoughts on the above issue?

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 3 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:189]: @phoracek Do you have any suggestions to update the wording for the above issue?

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 5 above)
Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:242]: Concur that this note shouldn't be mentioned in 4.15. CPU Hotplug will GA in 4.16.

Hot unplug not working in Windows is a limitation of that OS. Unsure if that warrants a RN over something in docs as it's a permanent limitation outside of our control.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 9]
```diff
@@ -190,73 +197,71 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
-//4.14 leave in per Dan K
-* You cannot run {VirtProductName} on a single-stack IPv6 cluster. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2193267[*BZ#2193267*])
 
 [discrete]
 [id="virt-4-15-ki-nodes"]
 ==== Nodes
-//4.14 Leave in per Simone
+//4.15 Check with Stu
+//Need link to engineering bug instead of docs bug
 * Uninstalling {VirtProductName} does not remove the `feature.node.kubevirt.io` node labels created by {VirtProductName}. You must remove the labels manually.
```
Comment (sjhala-ccs) [inline virt/release_notes/virt-4-15-release-notes.adoc:218]: @stu-gott Is there an engineering bug that I can link to for the above issue?

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: /cc @tiraboschi

I think HCO is the component that creates this label. Can you point @sjhala-ccs in the right direction?

Comment (tiraboschi) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: I think that it's added by node-labeller executed by virt-handler: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/node-labeller/node_labeller.go#L249

Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: @tiraboschi So the labels still need to be removed manually and this bug applies to 4.15? Can you provide a link to the actual bug? Thanks!

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: @sjhala-ccs Neither Virtualization nor Install, Upgrade and Operators components appear to have a relevant bug for this.

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: I'll need 24 hours to validate if this issue exists.

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: OK, it looks like those labels are not cleaned up. The RN is correct.

Comment (stu-gott) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: I've added a dev tracker to clean this up.

https://issues.redhat.com/browse/CNV-38543

Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:218]: Thanks. I will add that link to the RN.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 10]
```diff
@@ -186,6 +186,13 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
 
 [id="virt-4-15-bug-fixes"]
 == Bug fixes
+* Previously, the `windows-efi-installer` pipeline failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])
+
+* Previously, if you simultaneously cloned approximately 1000 VMs using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```
Comment (skopacz1) [inline virt/release_notes/virt-4-15-release-notes.adoc:191]: Suggestion to spell out the first instance of VM in this section:
```suggestion
* Previously, if you simultaneously cloned approximately 1000 virtual machines (VMs) using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
```

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 11]
```diff
@@ -194,73 +201,48 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
 ** As a workaround, use user accounts rather than service accounts because user account tokens are not bound to a specific pod.
 
-// Leave in for 4.14 per Stu; remove from 4.15 RNs and move content to CNV docs, per Avital
-* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2157951[*BZ#2157951*])
+//4.15 Keep this per Stu, ETA for fixing this issue is 4.18. 
+* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected.
```
Comment (skopacz1) [inline virt/release_notes/virt-4-15-release-notes.adoc:244]: Just curious, is there a reason that EMS is not in literals when the written out term is?
```suggestion
* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (`EMS`) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled RHEL 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 
```

Comment (skopacz1) [inline virt/release_notes/virt-4-15-release-notes.adoc:244]: I'd change this to the attribute, and possibly the "first mention" version as well:
```suggestion
* With the release of the link:https://access.redhat.com/errata/RHSA-2023:3722[RHSA-2023:3722] advisory, the TLS `Extended Master Secret` (EMS) extension (link:https://datatracker.ietf.org/doc/html/rfc7627[RFC 7627]) is mandatory for TLS 1.2 connections on FIPS-enabled {op-system-base-full} 9 systems. This is in accordance with FIPS-140-3 requirements. TLS 1.3 is not affected. 
```

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 12]
```diff
@@ -194,73 +201,48 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
...
-
-[discrete]
-[id="virt-4-15-ki-webconsole"]
-==== Web console
-//CNV-34771 from Ying
-* If you upgrade {product-title} 4.13 to 4.14 without upgrading {VirtProductName}, the Virtualization pages of the web console crash. (link:https://bugzilla.redhat.com/show_bug.cgi?id=OCPBUGS-22853[*OCPBUGS-22853*])
-
-** As a workaround, you must upgrade the {VirtProductName} Operator to 4.14 manually or set your subscription approval strategy to *Automatic*.
\ No newline at end of file
+** As a workaround, upgrade legacy OpenSSL clients to a version that supports TLS 1.3 and configure {VirtProductName} to use TLS 1.3, with the `Modern` TLS security profile type, for FIPS mode.
```
Comment (skopacz1) [inline virt/release_notes/virt-4-15-release-notes.adoc:248]: Update is the preferred term per our [term glossary](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#update):
```suggestion
** As a workaround, update legacy OpenSSL clients to a version that supports TLS 1.3 and configure {VirtProductName} to use TLS 1.3, with the `Modern` TLS security profile type, for FIPS mode.
```

**Review by skopacz1 — COMMENTED** [review]
Left a few comments but otherwise LGTM

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 11 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:244]: This is pre-existing content from 4.14.

Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc:244]: Pre-existing content. I am fine with updating it, but RHEL has been spelled out in the earlier sections.

File: virt/release_notes/virt-4-15-release-notes.adoc [hunk 13]
```diff
@@ -186,6 +186,13 @@ link:https://access.redhat.com/support/offerings/techpreview[Technology Preview
 
 [id="virt-4-15-bug-fixes"]
 == Bug fixes
+* Previously, the `windows-efi-installer` pipeline failed when started with a storage class that had the `volumeBindingMode` set to `WaitForFirstConsumer`. This fix removes the annotation in the `StorageClass` object that was causing the pipelines to fail. (link:https://issues.redhat.com/browse/CNV-32287[*CNV-32287*])
+
+* Previously, if you simultaneously cloned approximately 1000 virtual machines (VMs) using the provided data sources in the `openshift-virtualization-os-images` namespace, not all of the VMs moved to a running state. With this fix, you can clone a large number of VMs concurrently. (link:https://issues.redhat.com/browse/CNV-30083[*CNV-30083*])
+
+* Previously, you could not SSH into a VM when using the `networkType: OVNKubernetes` option in your `install-config.yaml` file. With this update, you can use a `NodePort` service for VM SSH access. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```
Comment (phoracek) [inline virt/release_notes/virt-4-15-release-notes.adoc:193]:
```suggestion
* Previously, you could not SSH into a VM over a NodePort Service and its associated FQDN displayed in the UI when using OVNKubernetes. With this update, you can configure Console so it shows a valid accessible endpoint for SSH NodePort Services. (link:https://issues.redhat.com/browse/CNV-24889[*CNV-24889*])
```

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 3 above)
Comment (phoracek) [reply virt/release_notes/virt-4-15-release-notes.adoc:193]: I added a suggestion for an alternative text.

File: virt/release_notes/virt-4-15-release-notes.adoc (hunk 13 above)
Comment (sjhala-ccs) [reply virt/release_notes/virt-4-15-release-notes.adoc]: Updated per your suggestion

**Pull request merged**